comptests/prov/test_prov_k8s_cortx_cold_upgrade.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+"""Provisioner Component level test cases for K8s CORTX Software Cold Upgrade."""
+
+import logging
+import os
+import time
+import queue
+from threading import Thread
+import pytest
+
+from commons import commands
+from commons import constants as cons
+from commons.helpers.pods_helper import LogicalNode
+from commons.utils import assert_utils
+from config import CMN_CFG, PROV_CFG, PROV_TEST_CFG
+from libs.ha.ha_common_libs_k8s import HAK8s
+from libs.prov.prov_k8s_cortx_deploy import ProvDeployK8sCortxLib
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TestProvK8CortxColdUpgrade:
+    """
+    This class contains Provisioner Component level test cases for K8s
+    CORTX Software Cold Upgrade.
+    """
+
+    @classmethod
+    def setup_class(cls):
+        """Setup class"""
+        LOGGER.info("STARTED: Setup Module operations")
+        cls.cortx_all_image = os.getenv("CORTX_ALL_IMAGE", None)
+        cls.cortx_rgw_image = os.getenv("CORTX_RGW_IMAGE", None)   
+        cls.deploy_cfg = PROV_CFG["k8s_cortx_deploy"]
+        cls.prov_deploy_cfg = PROV_TEST_CFG["k8s_prov_cortx_deploy"]
+        cls.deploy_lc_obj = ProvDeployK8sCortxLib()
+        cls.num_nodes = len(CMN_CFG["nodes"])
+        cls.worker_node_list = []
+        cls.master_node_list = []
+        cls.host_list = []
+        cls.local_sol_path = cons.LOCAL_SOLUTION_PATH
+        for node in range(cls.num_nodes):
```

Comment:
You can optimize to 
for node in cls.num_nodes:
      node_obj = LogicalNode(hostname=node["hostname"],
                                   username=node["username"],
                                   password=node["password"])